app/parse.py

```diff
@@ -1,22 +1,109 @@
-from dataclasses import dataclass
+import csv
+from dataclasses import dataclass, astuple
+from time import sleep
 from urllib.parse import urljoin
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from selenium.common.exceptions import (
+    ElementNotInteractableException,
+    ElementClickInterceptedException,
+)
 
+PRODUCT_FIELDS = [
+    "title",
+    "description",
+    "price",
+    "rating",
+    "num_of_reviews",
+]
 
 BASE_URL = "https://webscraper.io/"
 HOME_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/")
+COMPUTERS_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/computers")
+LAPTOPS_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/computers/laptops")
+TABLETS_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/computers/tablets")
+PHONES_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/phones")
+TOUCH_URL = urljoin(BASE_URL, "test-sites/e-commerce/more/phones/touch")
+
+URLS = {
+    "home": HOME_URL,
+    "computers": COMPUTERS_URL,
+    "laptops": LAPTOPS_URL,
+    "tablets": TABLETS_URL,
+    "phones": PHONES_URL,
+    "touch": TOUCH_URL,
+}
 
 
 @dataclass
 class Product:
-    title: str
-    description: str
-    price: float
-    rating: int
-    num_of_reviews: int
+    title: str = None
+    description: str = None
+    price_in_usd: float = None
+    rating: int = None
+    num_of_reviews: int = None
+
+
+def write_to_csv(products: list[str], output_csv_path: str) -> None:
+    with open(output_csv_path, "w") as csv_file:
+        writer = csv.writer(csv_file)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows(([astuple(product) for product in products]))
+
+
+def accept_cookies(driver: webdriver.Chrome) -> None:
+    accept_cookies_button = driver.find_elements(
+        By.CLASS_NAME, "acceptCookies"
+    )
+    if accept_cookies_button:
+        sleep(1)
```

Comment:
https://selenium-python.readthedocs.io/waits.html#explicit-waits
try to use WebDriverWait